src/components/AnnotationViewer.vue

```diff
@@ -2159,6 +2159,245 @@ export default class AnnotationViewer extends Vue {
+    const intersections = this.findAllIntersections(polygon, newLine);
+
+    // Group intersections by line segment
+    const intersectionsBySegment = intersections.reduce(
+      (acc, intersection) => {
+        if (!acc[intersection.lineSegmentIndex]) {
+          acc[intersection.lineSegmentIndex] = [];
+        }
+        acc[intersection.lineSegmentIndex].push(intersection);
+        return acc;
+      },
+      {} as Record<number, typeof intersections>,
+    );
+
+    // Find first and last intersection
+    let firstIntersection: (typeof intersections)[0] | null = null;
+    let lastIntersection: (typeof intersections)[0] | null = null;
+
+    // Find the first valid intersection
+    for (let i = 0; i < newLine.length - 1; i++) {
+      const segmentIntersections = intersectionsBySegment[i];
+      if (segmentIntersections?.length > 0) {
+        firstIntersection = segmentIntersections[0];
+        break;
+      }
+    }
+
+    // Find the last valid intersection
+    for (let i = newLine.length - 2; i >= 0; i--) {
+      const segmentIntersections = intersectionsBySegment[i];
+      if (segmentIntersections?.length > 0) {
+        lastIntersection =
+          segmentIntersections[segmentIntersections.length - 1];
+        break;
+      }
+    }
+
+    if (!firstIntersection || !lastIntersection) {
+      return annotation;
+    }
+
+    // Determine if we need to reverse the line points based on intersection order
+    const shouldReverseLine = firstIntersection.index > lastIntersection.index;
+
+    // Get the relevant line points
+    let linePoints: IGeoJSPosition[];
+    if (shouldReverseLine) {
+      linePoints = newLine
+        .slice(
+          firstIntersection.lineSegmentIndex + 1,
+          lastIntersection.lineSegmentIndex + 1,
+        )
+        .reverse();
+    } else {
+      linePoints = newLine.slice(
+        firstIntersection.lineSegmentIndex + 1,
+        lastIntersection.lineSegmentIndex + 1,
+      );
+    }
+
+    // Create new coordinates array
+    const newCoordinates: IGeoJSPosition[] = [
+      ...polygon.slice(
+        0,
+        Math.min(firstIntersection.index, lastIntersection.index),
+      ),
+      shouldReverseLine ? lastIntersection.point : firstIntersection.point,
+      ...linePoints,
+      shouldReverseLine ? firstIntersection.point : lastIntersection.point,
+      ...polygon.slice(
+        Math.max(firstIntersection.index, lastIntersection.index),
+      ),
+    ];
+
+    // Ensure the polygon is closed
+    if (
+      !this.pointsEqual(
+        newCoordinates[0],
+        newCoordinates[newCoordinates.length - 1],
+      )
+    ) {
+      newCoordinates.push({ ...newCoordinates[0] });
+    }
+
+    return {
+      ...annotation,
+      coordinates: newCoordinates,
+    };
+  }
+
+  // Helper function to check if two points are equal
+  pointsEqual(a: IGeoJSPosition, b: IGeoJSPosition): boolean {
+    return Math.abs(a.x - b.x) < 1e-10 && Math.abs(a.y - b.y) < 1e-10;
+  }
```

Comment:
I feel like these functions should go in a `utils` file, especially considering the size of this file